core/node/main/config.cpp

```diff
@@ -0,0 +1,277 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "config.hpp"
+
+#include <fstream>
+#include <iostream>
+
+#include <spdlog/fmt/fmt.h>
+#include <boost/asio.hpp>
+#include <boost/filesystem.hpp>
+#include <boost/program_options.hpp>
+
+namespace fc::node {
+
+  namespace {
+    constexpr int kDefaultPort = 2000;
+    constexpr std::string_view kDefaultConfigFileName = "default.cfg";
+
+    const std::string &getLocalIP() {
+      static const std::string a = [] {
+        boost::asio::io_context io;
+        boost::asio::ip::tcp::resolver resolver(io);
+        boost::asio::ip::tcp::resolver::query query(
+            boost::asio::ip::host_name(), "");
+        boost::asio::ip::tcp::resolver::iterator it = resolver.resolve(query);
```

Comment:
Throws exception sometimes
```suggestion
        boost::system::error_code ec;
        boost::asio::ip::tcp::resolver::iterator it = resolver.resolve(query, ec);
```